src/Microsoft.AspNetCore.Mvc.Formatters.Json/MvcJsonOptionsExtensions.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public static class MvcJsonOptionsExtensions
+    {
+        /// <summary>
+        /// Camel case property names.
+        /// </summary>
+        /// <param name="options"><see cref="MvcJsonOptions"/></param>
+        /// <param name="processDictionaryKeys">If true will camel case dictionary keys.</param>
+        /// <returns><see cref="MvcJsonOptions"/> with camel case settings</returns>
+        public static MvcJsonOptions UseCamelCasing(this MvcJsonOptions options, bool processDictionaryKeys)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            
+            if (options.SerializerSettings.ContractResolver.GetType() != typeof(DefaultContractResolver))
+            {
+                options.SerializerSettings.ContractResolver = new DefaultContractResolver
+                {
+                    NamingStrategy = new CamelCaseNamingStrategy
+                    {
+                        ProcessDictionaryKeys = processDictionaryKeys
+                    }
+                };
+                return options;
+            }
+
+            var defaultResolver = options.SerializerSettings.ContractResolver as DefaultContractResolver;
+
+            if (defaultResolver.NamingStrategy.GetType() == typeof(CamelCaseNamingStrategy))
+            {
+                defaultResolver.NamingStrategy.ProcessDictionaryKeys = processDictionaryKeys;
+            }
+            else
+            {
+                defaultResolver.NamingStrategy = new CamelCaseNamingStrategy
+                {
+                    ProcessDictionaryKeys = processDictionaryKeys
+                };
+            }
```

Comment:
I think the least surprising thing to do here is to just replace the contract resolver in **all** cases. Someone who wants to do custom things with the contract resolver will already understand how to do what they want. I don't see a reason to make this more complicated.

/cc @JamesNK  in case he has thoughts.

Reply:
There are some other settings on `DefaultContractResolver` so completely replacing it would overwrite those.

Reply:
Thank you for the reply @JamesNK , the only thing that should be touched is the naming strategy then?

Reply:
Ideally. I wonder if these methods should just throw if the type isn't `DefaultContractResolver`. 99% of people will always have type that assigned.

Reply:
@rynowak what do you think about throwing invalid operation exception if the type isn't DefaultContractResolver?